src/AutoRest.CSharp/Common/AutoRest/Plugins/Configuration.cs

```diff
@@ -132,8 +145,49 @@ public enum UnreferencedTypesHandlingOption
             _intrinsicTypesToTreatEmptyStringAsNull.UnionWith(additionalIntrinsicTypesToTreatEmptyStringAsNull);
         }
 
+        private static string? DownloadLatestContract(string projectFolder)
+        {
+            if (AzureArm || Generation1ConvenienceClient)
+            {
+                return null;
+            }
+
+            int sdkFolderIndex = projectFolder.LastIndexOf("\\sdk\\", StringComparison.InvariantCultureIgnoreCase);
+            if (sdkFolderIndex == -1)
+            {
+                return null;
+            }
+
+            string rootFolder = projectFolder.Substring(0, sdkFolderIndex);
+            var scriptPath = Path.Join(rootFolder, "eng", "common", "scripts", "Download-Latest-Contract.ps1");
+            if (File.Exists(scriptPath))
+            {
+                string projectDirectory = projectFolder.EndsWith("src") ? Path.GetFullPath(Path.Join(projectFolder, "..")) : projectFolder;
+                var scriptStartInfo = new ProcessStartInfo("pwsh", $"-ExecutionPolicy ByPass {scriptPath} {projectDirectory}")
+                {
+                    RedirectStandardOutput = false,
+                    RedirectStandardError = false,
+                    CreateNoWindow = true,
+                    UseShellExecute = false,
+                    WorkingDirectory = rootFolder
+                };
+                var scriptProcess = Process.Start(scriptStartInfo);
+                if (scriptProcess != null)
+                {
+                    scriptProcess.WaitForExit();
+
+                    string projectName = new DirectoryInfo(projectDirectory).Name;
+                    string relativeProject = projectFolder.Substring(sdkFolderIndex);
+                    return Path.GetFullPath(Path.Join(rootFolder, "..", "sparse-spec", "sdk", projectName, relativeProject));
+                }
+            }
+
+            return null;
+        }
+
         private static string? _outputFolder;
         public static string OutputFolder => _outputFolder ?? throw new InvalidOperationException("Configuration has not been initialized");
+        public static string? ExistingProjectFolder { get; internal set; }
```

Comment:
should this be `private set`? We should not change any configuration outside this class.